fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImplV2Spec.groovy

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.druid.client.impl
+
+import com.yahoo.bard.webservice.druid.client.DruidServiceConfig
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import spock.lang.Specification
+
+import org.asynchttpclient.Response
+
+import java.util.function.Supplier;
+
+class AsyncDruidWebServiceImplV2Spec extends Specification {
+    def "Make sure X-Druid-Response-Context and status-code are mreged into existing JsonNode"() {
+        given:
+        Response response = Mock(Response)
+        response.getResponseBody() >>
+                """
+                    {"k1":"v1"}
+                """
```

Comment:
Yes, why should it be an array and not an object? Are you saying that Druid's response is a top-level array?